projects/sonar/src/app/record/document/detail/detail.component.html

```diff
@@ -214,44 +199,26 @@ <h5 id="other-files" class="mt-5" translate>Other files</h5>
     <hr class="mb-4 mt-0">
     <div class="row">
       <ng-container *ngFor="let file of filteredFiles; let first = first">
-        <div class="text-center col-lg-2 mb-4" *ngIf="!first">
-          <ng-template let-thumbnail [ngTemplateOutletContext]="{ $implicit: getThumbnail(file) }"
-            [ngTemplateOutlet]="t" #t>
-            <a href (click)="$event.preventDefault(); previewFileKey = file.key; previewModal.show()" *ngIf="thumbnail">
-              <img [src]="thumbnail" class="img-thumbnail img-fluid">
-            </a>
-          </ng-template>
-          <p class="my-1 small">{{ file.label || file.key }}</p>
-          <p class="m-0">
-            <a href (click)="$event.preventDefault(); previewFileKey = file.key; previewModal.show()">
-              <i class="fa fa-eye mr-2"></i>
-            </a>
-            <a [href]="'/documents/' + record.pid + '/files/' + file.key" target="_blank">
-              <i class="fa fa-download"></i>
-            </a>
-          </p>
+        <div class="col-lg-2 mb-5" *ngIf="!first">
+          <sonar-document-file [file]="file" (previewClicked)="showPreview($event)"></sonar-document-file>
```

Comment:
Can any type of file can be previewed?

Reply:
No, only files that have a built-in previewer in invenio: https://invenio-previewer.readthedocs.io/en/latest/usage.html#bundled-previewers.